src/test/java/ro/msg/learning/shop/FindingLocationsStrategyTest.java

```diff
@@ -0,0 +1,71 @@
+package ro.msg.learning.shop;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import ro.msg.learning.shop.domain.*;
+import ro.msg.learning.shop.service.ProductService;
+import ro.msg.learning.shop.service.StockService;
+import ro.msg.learning.shop.service.findingLocatonsStrategy.ProductLocation;
+import ro.msg.learning.shop.service.findingLocatonsStrategy.SingleLocationStrategy;
+
+import java.math.BigDecimal;
+import java.util.*;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class FindingLocationsStrategyTest {
+
+    @Mock
+    private StockService stockService;
+
+    @Mock
+    private ProductService productService;
+
+    @InjectMocks
+    private SingleLocationStrategy singleLocationStrategy;
+
+    @Test
+    void testFindingLocationsStrategy() {
+        ProductCategory productCategory = new ProductCategory("IT", "Product From IT Category");
+        Supplier supplier1 = new Supplier("DELL");
+        Supplier supplier2 = new Supplier("HP");
+        Product product1 = new Product("Notebook Basic 15", "Notebook Basic", BigDecimal.valueOf(956),
+                2000.5, productCategory, supplier2,
+                "https://sapui5.hana.ondemand.com/test-resources/sap/ui/documentation/sdk/images/HT-1000.jpg");
+        Product product2 = new Product("Notebook Basic 15", "Dell Latitude", BigDecimal.valueOf(500),
+                1500.7, productCategory, supplier1,
+                "https://sapui5.hana.ondemand.com/test-resources/sap/ui/documentation/sdk/images/HT-1000.jpg");
+        when(productService.getById(1L)).thenReturn(product1);
+        when(productService.getById(2L)).thenReturn(product2);
+
+        Location location1 = new Location("Altex", "Romania", "Cluj-Napoca", "Cluj", "Memorandumului");
+        Location location2 = new Location("Flanco", "Romania", "Piatra Neamt", "Neamt", "9 Mai");
+        Stock stock1 = new Stock(product1, location1, 3);
```

Comment:
- We dont name variables 1, 2 etc .
- these could have been instance variables which are initialized in a before/before each method